tree/ntuple/v7/test/ntuple_bulk.cxx

```diff
@@ -0,0 +1,32 @@
+#include "ntuple_test.hxx"
+
+TEST(RNTupleBulk, Basics)
+{
+   FileRaii fileGuard("test_ntuple_bulk_basics.root");
+   {
+      auto model = RNTupleModel::Create();
+      auto fldInt = model->MakeField<int>("int");
+      auto writer = RNTupleWriter::Recreate(std::move(model), "ntpl", fileGuard.GetPath());
+      for (int i = 0; i < 20; ++i) {
+         *fldInt = i;
+         writer->Fill();
+      }
+   }
+
+   auto reader = RNTupleReader::Open("ntpl", fileGuard.GetPath());
+   // TODO(jblomer): find a better way to expose the GenerateBulk method of the target field
+   auto fieldZero = reader->GetModel()->GetFieldZero();
+   std::unique_ptr<RFieldBase::RBulk> bulk;
+   for (auto &f : *fieldZero) {
+      if (f.GetName() != "int")
+         continue;
+      bulk = std::make_unique<RFieldBase::RBulk>(f.GenerateBulk());
+   }
+
+   auto mask = std::make_unique<bool[]>(10);
+   memset(mask.get(), 1, 10);
+   auto intArr = static_cast<int *>(bulk->ReadBulk(RClusterIndex(0, 0), mask.get(), 10));
+   for (int i = 0; i < 10; ++i) {
+      EXPECT_EQ(i, intArr[i]);
+   }
+}
```

Comment:
it would be good to add a test that reads a collection. it's good to test that the resulting array elements are all contained in a flattened slab of memory, which is something RDF might rely on in the future.

Reply:
Yep, that can be done after adding the `ReadBulk` version for `RRVecField`. I wanted to do this in a follow-up PR but I can also do it here.